### ISSUE TITLE
FancyField Dropdown styles break when mixing 'className' and 'style' attributes

### ISSUE THREAD

**Issue by peterkle** [xogroup/union#260]
Context
===

- [x] Visual bug?
- [ ] Runtime bug?
- [ ] Build bug?

### *package name(s) and versions*

`"@xo-union/tk-component-fancy-fields": "^0.1.0"`

Issue
===

The API for FancyField Dropdown seems to recommend using the `style` attribute to customize components. The problem with this is:

1. This requires passing in a JS object, which can bloat the jsx with styles that would be better placed in separate sass files:
e.g.: 
```
<Dropdown
   onSelect={onChange}
  label={role}
  value={role}
  name="role"
  style={{ width: '10rem', textTransform: 'capitalize', backgroundColor: 'rgba(0,0,0,0)' }}
>
```

2. If one attempts to use `className` instead, the encoded styles are wiped out, instead of being appended to.

3. CSS modules depend on using `className`, and doing it this way complicates the process because `className`'s are exported as strings, while `style` requires an object. They do not co-exist nicely.

**Comment by GeorgeTaveras1231** [comment]
@peterkle I opened #306 to propose a standard for allowing style overrides. If approved by it should fix this issue.

**Comment by GeorgeTaveras1231** [comment]
@peterkle I'm unsure about what the desired behavior is (Which we need if we want to address before the implementation of a standard such as the one documented in #306). Should it be that passing a `className` prop combines the default `className` with the new one?

**Comment by peterkle** [comment]
>  Should it be that passing a className prop combines the default className with the new one?

@GeorgeTaveras1231 yes, that would be consistent with other implementations. I also support #306.

**Comment by GeorgeTaveras1231** [comment]
In that case, I'll mark it as `level:starter` and change the description.

Thanks for supporting it @peterkle, can you 👍 it?
